File: решения/cs/Classes/task8258example/Comparator8258.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Classes.Task8258example
{
    public class Comparator8258 : Comparer<int>
    {
        public override int Compare(int x, int y)
        {
            int d1 = Math.Abs(x - 6);
            int d2 = Math.Abs(y - 6);
            if (d1 > d2)
            {
                return 1;
            }
            else if (d1 == d2)
            {
                return 0;
            }
            else {
                return -1;
            }
        }
    }
}

File: решения/java/src/Lab15/task8258example/Comparator8258.java
package Lab15.task8258example;

import java.util.Comparator;

/**
 * @author akryukov
 *         18.07.2017
 */
public class Comparator8258 implements Comparator<Integer> {
    @Override
    public int compare(Integer o1, Integer o2) {
        int d1 = Math.abs(o1 - 6);
        int d2 = Math.abs(o2 - 6);
        if(d1 > d2){
            return 1;
        } else if (d1 == d2) {
            return 0;
        } else {
            return -1;
        }
    }
}
